device_smi/cpu.py

```diff
@@ -2,44 +2,46 @@
 import platform
 import subprocess
 
-from .base import BaseDevice, BaseInfo
+from .base import BaseDevice, BaseInfo, BaseMetrics
 
 
 class CPUInfo(BaseInfo):
     pass # TODO extend for cpu
 
+class CPUMetrics(BaseMetrics):
+    pass
+
 class CPUDevice(BaseDevice):
     def __init__(self, index: int = 0):
         super().__init__(index)
+        self._info = self.info()
 
-    def info(self) -> CPUInfo:
-        def cpu_utilization():
-            # check if is macos
-            if platform.system() == 'Darwin':
-                result = subprocess.run(['top', '-l', '1', '-stats', 'cpu'], stdout=subprocess.PIPE)
-                output = result.stdout.decode('utf-8')
-
-                # CPU usage: 7.61% user, 15.23% sys, 77.15% idle
-                for line in output.splitlines():
-                    if line.startswith('CPU usage'):
+    def _cpu_utilization(self):
```

Comment:
Rename to `_utilization`. The fn is already inside CpuMetrics so no need to double label with _cpu